provider/oci/oci.go

```diff
@@ -113,27 +115,61 @@ func (p *OCIProvider) zones(ctx context.Context) (map[string]dns.ZoneSummary, er
 
 	log.Debugf("Matching zones against domain filters: %v", p.domainFilter.Filters)
 	var page *string
-	for {
-		resp, err := p.client.ListZones(ctx, dns.ListZonesRequest{
-			CompartmentId: &p.cfg.CompartmentID,
-			ZoneType:      dns.ListZonesZoneTypePrimary,
-			Page:          page,
-		})
-		if err != nil {
-			return nil, errors.Wrapf(err, "listing zones in %q", p.cfg.CompartmentID)
-		}
 
-		for _, zone := range resp.Items {
-			if p.domainFilter.Match(*zone.Name) && p.zoneIDFilter.Match(*zone.Id) {
-				zones[*zone.Name] = zone
-				log.Debugf("Matched %q (%q)", *zone.Name, *zone.Id)
-			} else {
-				log.Debugf("Filtered %q (%q)", *zone.Name, *zone.Id)
+	// set zone scope based on configuration flag
+	scope := dns.ListZonesScopeEnum(p.zoneScope)
+	if scope == "" {
+		// return zones with GLOBAL and PRIVATE scopes
+		scopes := dns.GetGetZoneScopeEnumValues()
+		for _, scope := range scopes {
+			for {
+				resp, err := p.client.ListZones(ctx, dns.ListZonesRequest{
+					CompartmentId: &p.cfg.CompartmentID,
+					ZoneType:      dns.ListZonesZoneTypePrimary,
+					Scope:         dns.ListZonesScopeEnum(scope),
+					Page:          page,
+				})
+				if err != nil {
+					return nil, errors.Wrapf(err, "listing zones in %q", p.cfg.CompartmentID)
+				}
+
+				for _, zone := range resp.Items {
+					if p.domainFilter.Match(*zone.Name) && p.zoneIDFilter.Match(*zone.Id) {
+						zones[*zone.Name] = zone
```

Comment:
This is potentially problematic, as the names of private zones are not unique, so someone could have more than one zone with the same name in the compartment. That's a bit of an edge case though.

It would be better to use the zone id as the index, although that would require changes elsewhere as well.